hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -563,13 +565,27 @@ public CalciteContextException newValidationError(SqlNode node, Resources.ExInst
             Object[] arguments = ResourceUtil.args(e);
             String identifier = (arguments != null && arguments.length > 0) ? String.valueOf(arguments[0]) : null;
             Mapping mapping = identifier != null ? iMapResolver.resolve(identifier) : null;
-            String sql = mapping != null ? SqlCreateMapping.unparse(mapping) : null;
+            String sql = mapping != null && hasMapAccess(identifier) ? SqlCreateMapping.unparse(mapping) : null;
             String message = sql != null ? ValidatorResource.imapNotMapped(e.str(), identifier, sql) : e.str();
             throw QueryException.error(SqlErrorCode.OBJECT_NOT_FOUND, message, exception, sql);
         }
         return exception;
     }
 
+    private boolean hasMapAccess(String map) {
```

Comment:
May I ask you to add docs to this method? The contract is not obvious from the name

Reply:
The description is too generic, I'd suggest adding some details about why was it added and the reasoning for such solution.

Reply:
this method is quite general. use whenever you need to check permissions without interrupting the process.

added more details to the doc. is that ok?